src/vm_cpu.rs

```diff
@@ -113,9 +118,12 @@ impl VmPmuState {
 
     /// Sets the internal state for started PMU counters.
     /// This is intended to be called after a successful SBI invocation for starting PMU counters.
-    pub fn set_started_counter(&mut self, counter_index: u64, _initial_value: Option<u64>) {
+    pub fn set_started_counter(&mut self, counter_index: u64, initial_value: Option<u64>) {
         if (counter_index as usize) < self.configured_counter_map.len() {
             self.started_counters |= 1 << counter_index;
+            if let Some(initial_value) = initial_value {
+                self.counter_values[counter_index as usize] = initial_value;
```

Comment:
Maybe I'm missing something, but why do we care what the VM sets as the initial value here? Won't we overwrite this anyway when we context switch out?

Reply:
Per our offline discussion, this is needed so that any initial value specified during the start doesn't get lost. There an similar case where the counter gets reset, but we'll definitely add a comment since it's counter intuitive.